DOCUMENTATION.md

```diff
@@ -1345,23 +1349,44 @@ The following are example configurations.
 
 ```javascript
 var clientConfig = new Config.ClientConfig();
-clientConfig.networkConfig.connectionAttemptLimit = 5;
+clientConfig.clusterName = 'hzCluster';
+
+clientConfig.networkConfig.cloudConfig.enabled = true;
+clientConfig.networkConfig.cloudConfig.discoveryToken = 'EXAMPLE_TOKEN';
 ```
 
-Its default value is `2`.
+To be able to connect to the provided IP addresses, you should use secure TLS/SSL connection between the client and members. Therefore, you should set an SSL configuration as described in the previous section.
+
+# 6. Client Connection Strategy
 
-## 5.6. Setting Connection Attempt Period
+Node.js client can be configured to connect to a cluster in an async manner during the client start and reconnecting
+after a cluster disconnect. Both of these options are configured via `ClientConnectionStrategyConfig`.
 
-Connection attempt period is the duration in milliseconds between the connection attempts defined by `ClientNetworkConfig.connectionAttemptLimit`.
+You can configure the client’s starting mode as async or sync using the configuration element `asyncStart`.
+When it is set to `true` (async), the behavior of `Client.newHazelcastClient()` call changes.
+It resolves a client instance without waiting to establish a cluster connection. In this case, the client rejects
+any network dependent operation with `ClientOfflineError` immediately until it connects to the cluster. If it is `false`,
+the call is not resolved and the client is not created until a connection with the cluster is established.
+Its default value is `false` (sync).
 
-The following are example configurations.
+You can also configure how the client reconnects to the cluster after a disconnection. This is configured using the
+configuration element `reconnectMode`; it has three options:
+
+* `OFF`:  Client rejects to reconnect to the cluster and triggers the shutdown process.
+* `ON`: Client opens a connection to the cluster in a blocking manner by not resolving any of the waiting invocations.
```

Comment:
nit: the name of this value seems weird to me. Both `ON` and `ASYNC` force the client to reconnect, so I'd rather change `ON` to something like `SYNC`. But I guess the same name is used in Java client, so it's problematic to change it.

Reply:
I think in all clients, we try to keep the naming of things consistent with the Java client.